technical-guide/developer/frontend.md

```diff
@@ -334,100 +334,120 @@ Ensure your development environment docker image is up to date.
 
-1. In a terminal on your host machine, navigate to the frontend folder, then run the next command:
+1. In a terminal _in your host machine_, navigate to the `frontend` folder, then run:
 
 ```bash
 # cd <repo>/frontend
-
 npx playwright test --ui
 ```
 
-> ❗**WARNING** It is important to be on the right folder `frontend` of the project or we may have silent errors trying to run the tests.
+> ⚠️ **WARNING:** It is important to be in the right folder (`frontend`) to launch the command above, or you may have errors trying to run the tests.
+
+> ❗️ **IMPORTANT**: You might need to [install Playwright's browsers and dependencies](https://playwright.dev/docs/intro) in your host machine with: `npx playwright install --with-deps`.
 
 ### How to write a test
 
-#### Page Object Model
+When writing integration tests, we are simulating user actions and events triggered by them, in other to mirror real-world user interactions. The difference with fully end-to-end tests is that here we are faking the backend by intercepting the network requests, so tests can run faster and more tied to the front-end.
+
+Keep in mind:
+
+- **Use Realistic User Scenarios:** Design test cases that mimic real user scenarios and interactions with the application.
+
+- **Simulate User Inputs**: Such as mouse clicks, keyboard inputs, form submissions, or touch gestures, using the testing framework's API. Mimic user interactions as closely as possible to accurately simulate user behavior.
 
-When conducting a significant number of tests, encountering repetitive code and common actions is typical.
-To address this issue, we recommend leveraging Page Object Models (POM).
+- **Intercept the network**: Playwright offers ways to fake network responses to API calls, websocket messages, etc. Remember that there is no backend here, so you will need to intercept every request made by the front-end app.
 
-Page Object Models allow us to consolidate information into a single class and encapsulate it.
+#### Page Object Model
+
+When writing a significant number of tests, encountering repetitive code and common actions is typical. To address this issue, we recommend leveraging **Page Object Models** (POM), which is a single class that encapsulates common locators, user interactions, etc.
 
-POMs do not necessarily refer to entire pages but can also represent specific regions of a page that are the focus of our tests. For example, we may have a POM for the login form, the footer of a complex page, or the projects section.
+POMs do not necessarily refer to entire pages but can also represent specific regions of a page that are the focus of our tests. For example, we may have a POM for the login form, or the projects section.
 
-In a POM, we define locators for page elements:
+In a POM, we can define locators in the constructor itself — remember that locators will be accessed when interacted with (with a `click()`, for instance) or when asserting expectations.
```

Comment:
TODO: Add that locators that are generic enough to be used in multiple tests, they should be part of the POM